Lib/test/test_ordered_dict.py

```diff
@@ -650,6 +650,43 @@ def test_free_after_iterating(self):
         support.check_free_after_iterating(self, lambda d: iter(d.values()), self.OrderedDict)
         support.check_free_after_iterating(self, lambda d: iter(d.items()), self.OrderedDict)
 
+    def test_kwargs_order(self):
+        # bpo-34320
+        od = self.OrderedDict([('a', 1), ('b', 2)])
+        od.move_to_end('a')
+        expected = list(od.items())
+
+        def fn(**kw):
+            return kw
+
+        res = fn(**od)
+        self.assertIsInstance(res, dict)
+        self.assertEqual(list(res.items()), expected)
+
+    def test_type_namespace_order(self):
+        # bpo-34320
+        od = self.OrderedDict([('a', 1), ('b', 2)])
+        od.move_to_end('a')
+        expected = list(od.items())
+
+        C = type('C', (), od)
+        self.assertEqual(list(C.__dict__.items())[:2], [('b', 2), ('a', 1)])
+
+    def test_dict_copy_order(self):
+        # Issue 34320
+        od = self.OrderedDict([('a', 1), ('b', 2)])
+        od.move_to_end('a')
+        expected = list(od.items())
+
+        copy = dict(od)
+        self.assertEqual(list(copy.items()), expected)
+
+    # Namespace preserves order by language spec from 3.6.
+    # Dict preserves order by language spec from 3.7.
+    # Dict preserves order by implementation from 3.6.
+    if sys.version_info < (3, 7):
+        test_dict_copy_order = support.cpython_only(test_dict_copy_order)
```

Comment:
In 3.7+ this is not needed. And I think that in <3.7 this test should be backported.

Reply:
Sorry, what do you mean concretely?

Reply:
Tests are ran with concrete Python version. `sys.version_info` is a constant in every branch. In branches master and 3.7 these two lines should be removed. In branch 3.6 the decorator should be added unconditionally. But the backport to 3.6 needs editing in any case (only the test should be backported), so I don't think this will add additional work.